src/container/srv_container.c

```diff
@@ -682,6 +699,13 @@ cont_prop_write(struct rdb_tx *tx, const rdb_path_t *kvs, daos_prop_t *prop,
 			rc = rdb_tx_update(tx, kvs, &ds_cont_prop_rp_pda,
 					   &value);
 			break;
+		case DAOS_PROP_CO_GLOBAL_VERSION:
+			d_iov_set(&value, &entry->dpe_val,
```

Comment:
This writes a 64-bit number into the rdb, whereas I think the intention was for cont_global_version to be a uint32_t in the database. And the function upgrade_cont_cb() appears to be explicitly attempting to use uint32_t type for the cont_global_version value for containers that get upgraded.  I filed issue https://daosio.atlassian.net/browse/DAOS-10929 with more information.